### PR TITLE
[MIRROR] New closets [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/79774
--------------------
closets_flat

closets_flat_1

<details>
  <summary>In-game look</summary>

image

</details>

## About The Pull Request

Resprites the existing closets and adds new ones.

TODO:

- [x] Resprite cabinets
- [x] Split wardrobe cabinets into unique closets
- [x] Add new sprites for some closets that use generic style as a fallback
- [x] Update the cursed closets and fridges for the new dimensions
- [x] Bio closet for medbay
- [x] Re-map some locker icons like chemistry, anasthetic...

## Why It's Good For The Game

Making them in the same style as the crates.

## Changelog

:cl:
fix: Emag overlay on lockers fixed
image: New locker sprites
image: Added new lockers
/:cl:

---

Also includes https://github.com/tgstation/tgstation/pull/73331

### CODE REVIEW

**Comment by HoodMilk** [comment]
Tall, they look good.

**Comment by ghost** [comment]
our modular lockers will need touching up... gimme a bit

Edit: this might be a **long** "bit". Feel free to merge without ours matching sprite-wise if it takes too long. Just for the love of god make sure any of the code changes don't brick our icons entirely.
image
It's happening though.

**Comment by vinylspiders** [comment]
> our modular lockers will need touching up... gimme a bit
> 
> Edit: this might be a **long** "bit". Feel free to merge without ours matching sprite-wise if it takes too long. Just for the love of god make sure any of the code changes don't brick our icons entirely. image It's happening though.

I can wait. I was just going to separate out the actually modular lockers from the base tg ones because we kind of have them all in one big overridden copy pasted override dmi right now.

**Comment by ghost** [comment]
> I can wait. I was just going to separate out the actually modular lockers from the base tg ones because we kind of have them all in one big overridden copy pasted override dmi right now.

Looking in the code, the overrides don't ACTUALLY override any TG closets. It's just a ton of duplicate, unused icons in the file used for fully modular lockers. Feel free to merge it if it works as-is, my PR will cull the modular .dmi of all but needed icons.

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 1]
```diff
@@ -298,7 +297,7 @@
 
 /obj/structure/closet/secure_closet/armory2
 	name = "armory ballistics locker"
-	icon_state = "armory" // SKYRAT EDIT ADDITION - NEW ICON ADDED IN peacekeeper_lockers.dm
+	icon_state = "tac"
```
Comment (vinylspiders) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm:300]:
```suggestion
	icon_state = "tac"
	icon_door = "armory_shotgun"
```
Make sure this matches tg completely

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 2]
```diff
@@ -311,7 +310,6 @@
 
 /obj/structure/closet/secure_closet/armory3
 	name = "armory energy gun locker"
-	icon_state = "armory" // SKYRAT EDIT ADDITION - NEW ICON ADDED IN peacekeeper_lockers.dm
 	req_access = list(ACCESS_ARMORY)
```
Comment (vinylspiders) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm:316]:
```suggestion
	icon_state = "tac"
	icon_door = "armory_energy"
	req_access = list(ACCESS_ARMORY)
```
Same here

File: modular_skyrat/modules/sec_haul/code/peacekeeper/peacekeeper_lockers.dm [hunk 3]
```diff
@@ -9,9 +9,13 @@
...
+	icon_state = "armory"
 
 /obj/structure/closet/secure_closet/armory2
 	icon = 'modular_skyrat/modules/sec_haul/icons/lockers/closet.dmi'
+	icon_state = "armory"
+	icon_door = null
 
 /obj/structure/closet/secure_closet/armory3
 	icon = 'modular_skyrat/modules/sec_haul/icons/lockers/closet.dmi'
+	icon_state = "armory"
```
Comment (vinylspiders) [inline modular_skyrat/modules/sec_haul/code/peacekeeper/peacekeeper_lockers.dm:21]:
```suggestion
	icon_state = "armory"
	icon_door = null
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]

**Comment by vinylspiders** [comment]
> > I can wait. I was just going to separate out the actually modular lockers from the base tg ones because we kind of have them all in one big overridden copy pasted override dmi right now.
> 
> Looking in the code, the overrides don't ACTUALLY override any TG closets. It's just a ton of duplicate, unused icons in the file used for fully modular lockers. Feel free to merge it if it works as-is, my PR will cull the modular .dmi of all but needed icons.

@OrionTheFox This will ship with the modular .dmi culled, so just import your new sprites into there once this gets merged!

**Review by vinylspiders — APPROVED** [review]
Looks good, nice work!

**Pull request merged**